### ISSUE TITLE
Errors in building opm-common

### ISSUE THREAD

**Issue by Rasha-Moussaoui** [OPM/opm-simulators#4875]
Upon building from the source, I got the below error message 

![2](https://github.com/OPM/opm-simulators/assets/112950806/8f2f3adc-d588-4af3-bf5a-7c924ee60260)

then I uninstalled the previously downloaded flow simulator from the binary [packages and retried to build the simulator, it gave the below error message. How can we fix this

![1](https://github.com/OPM/opm-simulators/assets/112950806/51a7bca5-63f3-4157-9a05-07d8d95fdc11)

**Comment by bska** [comment]
How much memory does your virtual machine have?  For what it's worth, `Summary.cpp` is rather large and you **might** be running out of memory for the `cc1plus` process.

**Comment by Rasha-Moussaoui** [comment]
It was 2 GB, I tried changing it to 16 GB and retrying and it worked! Thanks

**Comment by bska** [comment]
> It was 2 GB, I changed it to 16 GB and retried building and it worked!

Very good.  Please close the issue if you're satisfied that the problem has been resolved.

**Comment by Rasha-Moussaoui** [comment]
one more thing please. I gave the full path to the Flow executable but got this, what do you recommend

![3](https://github.com/OPM/opm-simulators/assets/112950806/24b58358-277b-44ed-ad13-5a785407bbcd)
![4](https://github.com/OPM/opm-simulators/assets/112950806/fb69ab6a-bad9-4bbf-b997-b0081ddaffb8)

**Comment by akva2** [comment]
unlike windows systems the current directory is not in the path. you have to use ./flow to run the binary in current directory.

**Comment by Rasha-Moussaoui** [comment]
what am I doing wrong?
![5](https://github.com/OPM/opm-simulators/assets/112950806/b1265c1f-f81a-458a-9b88-21b57c76aa3d)
![6](https://github.com/OPM/opm-simulators/assets/112950806/4cd42897-26dc-4a4c-9f87-08a6593dc299)

**Comment by akva2** [comment]
from the looks of it, nothing except giving the wrong path to the data case. 
```
~Desktop 
```
is the home directory of the 'Desktop' user. 
```
~/Desktop
```
is the Desktop folder in your home folder.

**Comment by akva2** [comment]
also i would do it the other way around
```
cd ~/Desktop/opm-data/spe1
~/Desktop/project/opm-simulators/build/bin/flow SPE1CASE1.DATA
```

**Comment by Rasha-Moussaoui** [comment]
Thank you so much! It worked!

**Comment by Rasha-Moussaoui** [comment]
when trying to use the ecl_summary, it says the command not found, how I can resolve this

**Comment by bska** [comment]
> when trying to use the `ecl_summary`, it says the command not found, how I can resolve this

That depends greatly on the specifics of what you're trying to accomplish.  If you're trying to generate a table of numbers for humans, we have the `summary` utility which, apart from a few minor differences in the column headings is a drop-in replacement for `ecl_summary`.  If you're trying to plot selected curves there is either [ResInsight](https://resinsight.org/) or [qsummary](https://github.com/OPM/qsummary/).  If you're trying to import result curves into an analysis engine you may want to look into the Python bindings.

In short, unless you tell us what you're trying to accomplish we can't really give any specific advice.

**Comment by Rasha-Moussaoui** [comment]
I was trying to apply the tutorial of SPE1 case on the website

![WhatsApp Image 2023-09-19 at 8 08 00 PM](https://github.com/OPM/opm-simulators/assets/112950806/92e8f428-b6bd-49d5-a57a-b4f02b27f803)

**Comment by akva2** [comment]
Yeah, I am afraid that tutorial page is outdated.

As part of the opm-common build you got a program called 'summary', likely
```
~/Desktop/project/opm-common/build/bin/summary
```
this can do the listing and table generation equivalent to the referenced ecl_summary, e.g.
```
~/Desktop/project/opm-common/build/bin/summary SPE1CASE WBHP:INJ WGIR:INJ WBHP:PROD WOPR:PROD WGPR:PROD
```
As for plotting there is either qsummary  or resinsight as @bska mentioned above. They are both superior to the old summary_plot script.
